Schedule Link Sharing

brelieu05 opened the pull request:
## Summary
Implemented Schedule Link Sharing

- User can share their schedule with a link
  - Currently, the sharing param is the db schedules UUID
- When the user opens the share link, they see the schedule, but it's only temporary and if they refresh it will disappear
  - User can press the normal save button to add it as a permanent schedule
  - It loads the user's schedule first -> adds + switches to shared schedule
- If invalid share link, snackbar will notify user so
 

<img width="313" height="168" alt="image" src="https://github.com/user-attachments/assets/ccd5e2b5-654f-47ff-a6fd-7cf110989eab" />
<img width="1040" height="379" alt="image" src="https://github.com/user-attachments/assets/c28ea6a8-7f67-49f2-b1a9-8fd502d67ffb" />
<img width="1680" height="1050" alt="image" src="https://github.com/user-attachments/assets/eff5e0ae-fdd7-4037-b9cf-989922474717" />
<img width="559" height="246" alt="image" src="https://github.com/user-attachments/assets/68e0716f-45ed-4a64-8787-ad27607bc878" />
<img width="559" height="1050" alt="image" src="https://github.com/user-attachments/assets/3a2cb1de-57ca-48b7-95ad-2c8f621519ba" />



## Test Plan

## Issues
How should I handle the multiple snackbars? Just one after everything loads or keep two to indicate when account schedules have loaded and when shared schedule has loaded? since they're separate things

Closes #1346

<!-- [Optional]
## Future Followup
-->

KevinWu098 commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -262,6 +262,38 @@ export const importScheduleWithUsername = async (username: string) => {
+    }
+
+    scheduleComponentsToggleStore.setState({ openLoadingSchedule: false });
+    return { imported: false, error: new Error('Failed to load shared schedule') };
```
nit: this would be more readable inverted

if (!isScheduleLoaded) {
	blah blah
	blah blah
}

// success logic

alexespejo commented on apps/antalmanac/src/components/dialogs/ShareSchedule.tsx:
```diff
@@ -0,0 +1,139 @@
+import { ContentCopy, Check } from '@mui/icons-material';
```
I think the "Share" button we use is typically associated with a dialog like this but to the average user the details of the link are unnecessary. I would just default to a copy link icon

alexespejo commented on apps/antalmanac/src/components/Calendar/Toolbar/ScheduleSelect/ScheduleSelect.tsx:
```diff
@@ -221,6 +222,7 @@ export function SelectSchedulePopover() {
                                         </Box>
 
                                         <Box display="flex" alignItems="center" gap={0.5}>
+                                            <ShareScheduleButton index={index} disabled={skeletonMode} />
```
Unsaved schedules won't have an ID so won't be shareable and you'll get an error. There are different ways you can approach it. 

1) You could just throw a warning that you need to save the schedule in order to save it 

2) Restructure the way you format the URL to not depend on the schedule id 
- You could scrape the section codes and just parse them in the client when you load the shared schedule. Might be a little bloated and buggy
- Could add a nice user experience by not having that small barrier (though not sure if it's worth the engineering time)

alexespejo commented on apps/antalmanac/src/backend/lib/rds.ts:
```diff
@@ -308,6 +308,67 @@ export class RDS {
         await tx.insert(customEvents).values(dbCustomEvents);
     }
 
+    /**
```
Instead of fetching the schedule ID you can include it in the payload when we initially fetch the schedule data scheout the `Schedules` type in [`schedule.ts`](https://github.com/icssc/AntAlmanac/blob/main/packages/types/src/schedule.ts)

KevinWu098 reviewed (dismissed):
I haven't read the code deeply, but I'll share a few thoughts/feedback before doing a deeper read later

1. Feature is definitely cool
2. the mergeShortCourseSchedules breaks because it adds a `(1)` so adding a shared schedule "Foo" becomes "Foo (1)" if the receiving user has another schedule named "Foo"
3. Anytime we ask someone to sign in, we should make it easy for them — give them a button
4. We should be more clear that this is a static copy
5. Relating to #4... do we have to be a static copy?  🌚

brelieu05 committed:
refactor: added back share api routes bc of merge

brelieu05 committed:
refactor: readablity

brelieu05 committed:
refactor: removed unnecessary call

brelieu05 committed:
feat: separate page schedule sharing

brelieu05 committed:
fix: can look at shared added tab

brelieu05 committed:
fix: search and map tabs redirection

brelieu05 committed:
fix: sign in

brelieu05 committed:
fix: add to schedule merge

brelieu05 committed:
fix: empty sharedschedule after adding bug

brelieu05 committed:
feat: exit button + loading

brelieu05 committed:
fix: non-logged-in user schedule add bug

alexespejo reviewed (changes requested):
1) I hate to be that guy but is there a way for you to consolidate the git diff so it only shows the file changes you made. If it's too much of a hassle it's whatevs we'll figure it out. 

2) Also can you restore the merge conflicts (resolving them might resolve the git diff issue) and may fix the staging deployment which seems to be breaking

<img width="783" height="61" alt="image" src="https://github.com/user-attachments/assets/cab4b573-5e5a-4d6e-971e-60f5410797e2" />

brelieu05 committed:
feat: schedule link sharing

brelieu05 committed:
Merge branch 'main' into 1346-sharable-course-schedule-links

brelieu05 committed:
fix: custom events test type error

brelieu05 committed:
fix: add schedule id issue + removed dialog for copy link

KevinWu098 committed:
ci: add `remove` and `unlock` sst commands (#1389)

KevinWu098 committed:
ci: validate sst commands (#1390)

KevinWu098 committed:
ci: sst slash commands (#1391)

KevinWu098 committed:
ci: sst commands

KevinWu098 committed:
feat: deploy command

KevinWu098 committed:
ci: replace chronorm with myrotvorets

alexespejo reviewed (changes requested):
Haven't looked at code too indepth yet but some bugs 🐞:

1) Unsaved changes 
Repro: Opening a shared link -> editing the schedule (removing/adding classes) -> "refresh" -> "accept unsaved changes" -> the changes persist 

The user either should be locked out of changing the shared schedule or lock unsaved changes from triggering 

2) Design

    1) Small detail I think it would be nice to preview the schedule name in the drop down button so that there isn't just some empty space 
    2) Maybe bold the name of the schedule being imported (e.g. "Viewing Shared Schedule: **2026 Winter**)
<img width="712" height="128" alt="image" src="https://github.com/user-attachments/assets/e0570af0-30ea-472d-b252-b658ecb9a6ed" />



3) Sharing a deleted schedule id "breaks" the user's data 

Repro: have two users open (1 and 2) -> (1) copy share link -> (1) edit and save the schedule -> (2) open the shared link -> recieve error message -> hit "go home" -> the user's data should appear empty 

Refreshing the page will fix the schedule but it would be offputting the the user to think their data has been wiped

alexespejo committed:
Broken Precommit Hook  (#1383)

Choollol committed:
Show RecruitmentBanner over HelpMenu and use MUI theme z-indexes (#1394)

brelieu05 committed:
feat: stop user from saving in sharedschedule

brelieu05 committed:
feat: disabled search

brelieu05 committed:
feat: remove class delete from sharedschedule

brelieu05 committed:
fix: peter portal icon

calebwongg committed:
feat: Missing Sections Warning (#1376)

Added a warning displaying the missing sections for a course on the added courses page.

When a user selects classes for a course, the required sections for a course are checked (e.g., lecture, lab, discussion). If any required section type is missing, the UI displays a warning so the user can correct their schedule.

alexespejo committed:
Fix: Duplicate users created on sign up (#1385)

brelieu05 committed:
refactor: disabled -> not show

brelieu05 committed:
fix: finals button

brelieu05 committed:
fix: schedule dropdown + bold schedule name

brelieu05 committed:
fix: go Home button blank issue

brelieu05 committed:
fix: invalid schedule loading twice issue

brelieu05 committed:
fix: remove search button in section table

brelieu05 committed:
fix: sharedschdule map search for class issue

KevinWu098 committed:
ci: sst deploy environment (#1406)

alexespejo commented on apps/antalmanac/src/components/Calendar/CourseCalendarEvent.tsx:
```diff
@@ -93,6 +94,8 @@ export const CourseCalendarEvent = ({ selectedEvent, scheduleNames, closePopover
     const quickSearch = useQuickSearch();
     const { isMilitaryTime } = useTimeFormatStore();
+    const location = useLocation();
+    const isSharedSchedulePage = location.pathname.startsWith('/share/');
```
chore: these two lines appear 11 times in the code base. I'd be good if there's a way to just make it a utility 

```ts
const location = useLocation();
const isSharedSchedulePage = location.pathname.startsWith('/share/');
 ```

alexespejo commented on apps/antalmanac/src/components/Calendar/Toolbar/ScheduleSelect/schedule-select-buttons/ShareScheduleButton.tsx:
```diff
@@ -0,0 +1,59 @@
+
+    const handleCopy = useCallback(async () => {
+        if (!sessionIsValid) {
+            enqueueSnackbar('Please sign in to share schedules', { variant: 'error' });
```
chore: we have an `AppStore` utility `AppStore.openSnackbar()I`

alexespejo replied:
thought: if the user is not signed in we can just disable the button with a tooltip instead of providing a notification

alexespejo commented on apps/antalmanac/src/components/Header/Signin.tsx:
```diff
@@ -275,18 +279,35 @@ export const Signin = () => {
+                            });
+                    }
+                } else {
+                    // Normal behavior: load schedules
```
nit: comment

alexespejo commented on apps/antalmanac/src/components/Header/Signin.tsx:
```diff
@@ -275,18 +279,35 @@ export const Signin = () => {
             if (savedUserID != null || sessionID !== null) {
-                void loadScheduleAndSetLoadingAuth(savedUserID ?? '', true);
+                if (isSharedSchedulePage) {
+                    const sessionToken = getLocalStorageSessionId() ?? '';
```
issue: `sessionID` is defined on line 279

alexespejo commented on apps/antalmanac/src/components/Header/Signin.tsx:
```diff
@@ -275,18 +279,35 @@ export const Signin = () => {
 
             if (savedUserID != null || sessionID !== null) {
-                void loadScheduleAndSetLoadingAuth(savedUserID ?? '', true);
+                if (isSharedSchedulePage) {
```
nit: since we abstract what the loadSchedule inside of `loadScheduleAndSetLoadingAuth` maybe provide a short comment explaining why we just validate the session and not load the data

alexespejo commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,381 @@
+                            </Typography>
+                            <Stack direction="row" spacing={1} alignItems="center">
+                                <Button variant="contained" onClick={handleAddToMySchedules}>
+                                    Add to My Schedules
```
issue: responsive design 

<img width="391" height="683" alt="image" src="https://github.com/user-attachments/assets/65dbcc04-1885-48d8-9d64-85cbad2b4c5d" />

alexespejo commented on apps/antalmanac/src/components/dialogs/ShareSchedule.tsx:
```diff
@@ -0,0 +1,123 @@
+        try {
+            await navigator.clipboard.writeText(shareUrl);
+            setCopied(true);
+            enqueueSnackbar('Link copied to clipboard!', { variant: 'success' });
```
chore: `AppStore.openSnackbar`

brelieu05 committed:
refactor: shared schedule hook

brelieu05 committed:
refactor: appstore opensnackbar

brelieu05 committed:
feat: disabled button if not signed in

brelieu05 committed:
nit: remove comment

brelieu05 committed:
refactor: implemented updateSession

brelieu05 committed:
refactor:  implemented updateSession + comment

brelieu05 committed:
fix: header responsiveness

brelieu05 committed:
fix: increased lazy loading offset (#1393)

dsnsgithub committed:
chore: add next-env.d.ts to .gitignore (#1395)

xgraceyan committed:
Update favicon (#1408)

Co-authored-by: Kevin Wu <kevinwu098@gmail.com>

AureliaSindhu committed:
fix: sat display (#1380)

brelieu05 committed:
Merge branch 'main' into 1346-sharable-course-schedule-links

alexespejo commented on apps/backend/src/routers/userData.ts:
```diff
@@ -0,0 +1,240 @@
+import { UserSchema } from '@packages/antalmanac-types';
```
issue: I got hella confused when I saw this. The backend was moved to `apps/antalmanac/src/backend`. I these routes should exist in there

brelieu05 replied:
yeah the routes I added are in both backends since I made these changes before the merge. Do you want me to remove the routes I made from the old backend?

brelieu05 committed:
refactor: remove old backend

alexespejo commented on apps/antalmanac/src/stores/AppStore.ts:
```diff
@@ -418,6 +432,8 @@ class AppStore extends EventEmitter {
 
     loadSkeletonSchedule(savedSchedule: ScheduleSaveState) {
         this.schedule.setSkeletonSchedules(savedSchedule.schedules);
+        // Set the current schedule index to match the saved schedule index
```
nit: remove comments

alexespejo commented on apps/antalmanac/src/components/ScheduleManagement/ScheduleManagement.tsx:
```diff
@@ -50,6 +53,12 @@ export function ScheduleManagement() {
             return;
         }
 
+        // Don't set search tab when on shared schedule page
```
nit: comments

alexespejo commented on apps/antalmanac/src/actions/ActionTypesStore.ts:
```diff
@@ -165,6 +169,12 @@ class ActionTypesStore extends EventEmitter {
     }
 
     async loadScheduleFromUnsavedActions() {
+        // Don't load unsaved actions when viewing a shared schedule (skeleton mode)
```
nit: comment

alexespejo commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -183,6 +191,16 @@ export async function autoSaveSchedule(
             },
         });
 
+        // Update schedule IDs in AppStore after saving
```
comment

alexespejo commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -183,6 +191,16 @@ export async function autoSaveSchedule(
+            for (let i = 0; i < Math.min(schedules.length, result.scheduleIds.length); i++) {
+                schedules[i].id = result.scheduleIds[i];
+            }
+            // Emit events to notify components of the ID updates
```
comment

alexespejo commented on apps/antalmanac/src/components/dialogs/ShareSchedule.tsx:
```diff
@@ -0,0 +1,123 @@
+import { ContentCopy, Check } from '@mui/icons-material';
```
issue: if unused remove file

alexespejo reviewed (changes requested):
Fix comments and I'll approve

brelieu05 committed:
chore: removed unnecessary comments + files

brelieu05 committed:
fix: prevent duplicate schedule names in mergeShortCourseSchedules

brelieu05 committed:
chore: remove TODO comment

alexespejo commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,413 @@
+import { useMediaQuery, useTheme, Stack, Alert, Button, Box, Typography, IconButton } from '@mui/material';
+import { LocalizationProvider } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFnsV2';
+import type { ScheduleSaveState, ShortCourseSchedule, ShortCourse } from '@packages/antalmanac-types';
```
issue: consolidate imports

alexespejo commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,413 @@
+        };
+    }, [scheduleId, setOpenLoadingSchedule]);
+
+    const handleExitSharedSchedule = useCallback(async () => {
```
issue: This function repeats a lot of logic with `handleAddToMySchedules` and `handleGoHome`. We should try to containerize our logic into reusable functions.

alexespejo reviewed (changes requested):
Couple comments. Approach seems sane.

brelieu05 committed:
chore: consolidated imports

brelieu05 committed:
chore: containerize repeating logic

alexespejo reviewed (changes requested):
I believe you have some merge conflicts

brelieu05 committed:
Merge branch 'main' into 1346-sharable-course-schedule-links

brelieu05 committed:
chore: eslint move import

brelieu05 committed:
chore: eslint useeffect warnings

brelieu05 committed:
chore: eslint useCallback dependency

brelieu05 committed:
fix: pass userInfo correctly to autoSaveSchedule

brelieu05 committed:
fix: add missing id to skeleton ShortCourseSchedule

brelieu05 committed:
fix: added missing id property to convertLegacySchedule

brelieu05 committed:
Merge branch 'main' into 1346-sharable-course-schedule-links

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -137,6 +137,14 @@ export const saveSchedule = async (
                     },
                 });
 
+                if (result && 'scheduleIds' in result && Array.isArray(result.scheduleIds)) {
```
Might wanna move this logic into a `setSchedules` function in `AppStore`, especially since it's duplicated in `autoSaveSchedule` too.

Also could consider changing the condition to `Array.isArray(result?.scheduleIds)`, just a little more concise.

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -186,6 +194,14 @@ export async function autoSaveSchedule(providerID: string, options: AutoSaveSche
             },
         });
 
+        if (result && 'scheduleIds' in result && Array.isArray(result.scheduleIds)) {
```
See [#r2772150367](https://github.com/icssc/AntAlmanac/pull/1357/changes#r2772150367)

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -204,13 +220,20 @@ export const mergeShortCourseSchedules = (
-        if (existingScheduleNames.has(schedule.scheduleName)) {
-            scheduleName = scheduleName + '(1)';
+        let scheduleName = `${importMessage}${schedule.scheduleName}`;
+        let counter = 1;
```
Small suggestion: Could move this logic into a separate util file and use it both here and in `Schedules.ts`. Might be better to use `Set` by default (?), so we could change `getNextScheduleName` to match.

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -283,6 +310,59 @@ export const importScheduleWithUsername = async (username: string) => {
+
+    const currentSchedules = AppStore.schedule.getScheduleAsSaveState();
+
+    mergeShortCourseSchedules(currentSchedules.schedules, [incomingSchedule], '(shared)-');
```
Might want to put the `(shared)-` prefix in a shared constant since it's also used in [`SharedSchedulePage`](https://github.com/icssc/AntAlmanac/pull/1357/changes#diff-bd0e9717702784099989654cb10966852a29b32bcb4796bc95acbde91d23682dR275).

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -283,6 +310,59 @@ export const importScheduleWithUsername = async (username: string) => {
+                },
+            });
+        } catch (err) {
+            console.error('Failed to auto-save after importing shared schedule:', err);
```
Should we also open a snackbar or something similar to tell the user something went wrong?

Choollol commented on apps/antalmanac/src/actions/AppStoreActions.ts:
```diff
@@ -283,6 +310,59 @@ export const importScheduleWithUsername = async (username: string) => {
+        } catch (err) {
+            console.error('Failed to auto-save after importing shared schedule:', err);
+        }
+    }
```
Is there anything specific that should happen if the user's session isn't valid?

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/ScheduleSelect/schedule-select-buttons/ShareScheduleButton.tsx:
```diff
@@ -0,0 +1,60 @@
+
+export function ShareScheduleButton({ index, disabled }: ShareScheduleButtonProps) {
+    const [copied, setCopied] = useState(false);
+    const { sessionIsValid } = useSessionStore();
```
```suggestion
    const sessionIsValid = useSessionStore(state => state.sessionIsValid);
```

So state changes only cause rerenders when needed.

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/CalendarToolbar.tsx:
```diff
@@ -159,7 +161,7 @@ export const CalendarToolbar = memo((props: CalendarPaneToolbarProps) => {
                             onClick={handleToggleFinals}
                             id={showFinalsSchedule ? 'finals-button-pressed' : 'finals-button'}
-                            disabled={skeletonMode}
+                            disabled={skeletonMode && !isSharedSchedulePage}
```
Small suggestion: Extract this into a shared variable for mobile and non-mobile.

Choollol replied:
Actually, question: Should the app be marked as being in skeleton mode when on the shared schedule page? Since there's a schedule loaded, it feels like it isn't in skeleton mode.

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/CalendarToolbar.tsx:
```diff
@@ -291,12 +294,14 @@ export const CalendarToolbar = memo((props: CalendarPaneToolbarProps) => {
-                        skeletonMode={skeletonMode}
-                        analyticsCategory={analyticsEnum.calendar}
-                    />
+                    {!isSharedSchedulePage && (
```
Small question: Should this disable the button rather than not render it at all? It would keep the layout consistent with the normal calendar, but not a big issue.

Choollol commented on apps/antalmanac/src/components/Calendar/CourseCalendarEvent.tsx:
```diff
@@ -152,22 +154,24 @@ export const CourseCalendarEvent = ({ selectedEvent, scheduleNames, closePopover
-                            <Delete fontSize="inherit" />
-                        </IconButton>
-                    </Tooltip>
+                    {!isSharedSchedulePage && (
```
Similar to https://github.com/icssc/AntAlmanac/pull/1357/changes#r2775833753

Choollol commented on apps/antalmanac/src/components/Map/Map.tsx:
```diff
@@ -224,9 +226,19 @@ export default function CourseMap() {
     const onBuildingChange = useCallback(
         (building?: ExtendedBuilding | null) => {
-            navigate(`/map?location=${building?.id}`);
+            if (isSharedSchedulePage) {
```
This code works fine, but issue is related: When clicking the location of a class from the popover in the calendar (`MapLink.tsx`), it goes out of the shared schedule page and some buggy stuff happens.

<img width="420" height="297" alt="Image" src="https://github.com/user-attachments/assets/3a98ed59-fd7a-4942-b1fb-2a78522c55c1" />

Choollol commented on apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx:
```diff
@@ -215,25 +238,50 @@ function ScheduleNoteBox() {
+        if (!skeletonSchedule.courses || skeletonSchedule.courses.length === 0) {
+            const regularCourses = AppStore.schedule.getCurrentCourses();
+            if (regularCourses.length > 0) {
+                return {
```
Small suggestion: Could give this some type safety by typing the return as `ShortCourseSchedule` and adding `id: undefined` to the returned object below.

Choollol commented on apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx:
```diff
@@ -215,25 +238,50 @@ function ScheduleNoteBox() {
 
     const sectionsByTerm: [string, string[]][] = useMemo(() => {
-        const result = skeletonSchedule.courses.reduce(
+        const courses = skeletonSchedule?.courses || [];
```
Small question: When would `skeletonSchedule` be nullish? I'm just asking because null/undefined aren't part of its type. Did you run into problems with this in your own testing?

Choollol commented on apps/antalmanac/src/components/ScheduleManagement/ScheduleManagementTab.tsx:
```diff
@@ -12,12 +13,77 @@ interface ScheduleManagementTabProps {
+            <Tab
+                id={tab.id}
+                icon={tab.icon}
+                iconPosition={isMobile ? 'top' : 'start'}
```
Extract shared props between this component and the typical non-shared-page component (at the very bottom). Mostly `iconPosition` and `sx`, I think the others are fine.

Choollol commented on apps/antalmanac/src/hooks/useIsSharedSchedulePage.tsx:
```diff
@@ -0,0 +1,10 @@
+import { useLocation } from 'react-router-dom';
+
```
We could have an extra layer of abstraction and create an additional `useIsReadOnlyView` hook or something similar that uses `useIsSharedSchedulePage`. This way, we wouldn't need to add conditions everywhere if we add an extra feature that needs to disable deleting/editing.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+            </Stack>
+        </Split>
+    );
+}
```
This is a lot of duplication with the `MobileHome` and `DesktopHome` components. Maybe we could just import those components and use them here?

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+    const theme = useTheme();
+    const navigate = useNavigate();
+    const { scheduleId } = useParams<{ scheduleId: string }>();
+    const { sessionIsValid } = useSessionStore();
```
```suggestion
    const sessionIsValid = useSessionStore((state) => state.sessionIsValid);
```

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+    const navigate = useNavigate();
+    const { scheduleId } = useParams<{ scheduleId: string }>();
+    const { sessionIsValid } = useSessionStore();
+    const { setOpenLoadingSchedule } = scheduleComponentsToggleStore();
```
```suggestion
    const setOpenLoadingSchedule = scheduleComponentsToggleStore((state) => state.setOpenLoadingSchedule);
```

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+            console.error('Error exiting shared schedule:', err);
+            setOpenLoadingSchedule(false);
+            navigate('/');
+        }
```
Can move `setOpenLoadingSchedule(false);` and `navigate('/');` outside the try-catch.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+
+            if (AppStore.getSkeletonMode()) {
+                AppStore.exitSkeletonMode();
+            }
```
Tiny nitpick: These can be moved outside the try-catch.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                    scheduleId,
+                });
+
+                const scheduleSaveState = {
```
Tiny nitpick: Could type this as `ScheduleSaveState`.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                    courses: [] as ShortCourse[],
+                    customEvents: [] as RepeatingCustomEvent[],
+                    scheduleNote: '',
+                } as ShortCourseSchedule;
```
I think 
```ts
const emptyScheduleData: ShortCourseSchedule = {
    scheduleName: defaultTerm.shortName.replaceAll(' ', '-'),
    courses: [] as ShortCourse[],
    customEvents: [] as RepeatingCustomEvent[],
    scheduleNote: '',
    id: undefined,
};
```
would be safer.

Could we also replace defaultTerm.shortName.replaceAll(' ', '-') with AppStore.getDefaultScheduleName()?

Optional suggestion: This has a lot of logic that's shared with `Schedules.ts`'s constructor, you could extract it into a utility function.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                courses: [] as ShortCourse[],
+                customEvents: [] as RepeatingCustomEvent[],
+                scheduleNote: '',
+            } as ShortCourseSchedule;
```
I think 
```ts
const emptyScheduleData: ShortCourseSchedule = {
    scheduleName: defaultTerm.shortName.replaceAll(' ', '-'),
    courses: [] as ShortCourse[],
    customEvents: [] as RepeatingCustomEvent[],
    scheduleNote: '',
    id: undefined,
};
```
would be safer.

Also, I think `defaultTerm.shortName.replaceAll(' ', '-')` could be replaced with `AppStore.getDefaultScheduleName()`?

Optional suggestion: This has a lot of logic that's shared with `Schedules.ts`'s constructor, you could extract it into a utility function.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                    scheduleIndex: 0,
+                };
+                await AppStore.loadSchedule(emptySchedule);
+            }
```
This is a lot of logic that's shared with `handleLoadSchedule`, maybe could replace this with a call to `handleLoadSchedule`.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+
+            if (AppStore.getSkeletonMode()) {
+                AppStore.exitSkeletonMode();
+            }
```
This looks like it could be extracted into a reusable function, maybe `beginLoadingSchedule` or something.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+
+            if (AppStore.getSkeletonMode()) {
+                AppStore.exitSkeletonMode();
+            }
```
See https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776575421

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                    throw new Error('No session token available');
+                }
+
+                await handleLoadSchedule(sessionToken);
```
This also looks like it could be extracted into a separate function, maybe loadSessionSchedule or something like that.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                throw new Error('No session token available');
+            }
+
+            await handleLoadSchedule(sessionToken);
```
See https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776579369

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+                AppStore.skeletonMode = true;
+                AppStore.emit('skeletonModeChange');
+                setScheduleName(sharedSchedule.scheduleName);
+                setOpenLoadingSchedule(false);
```
See https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776065343, though this applies to only `setOpenLoadingSchedule(false);`.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+            }
+
+            setOpenLoadingSchedule(false);
+            navigate('/');
```
See https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776065343

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+            await handleLoadSchedule(sessionToken);
+
+            setOpenLoadingSchedule(false);
+            navigate('/');
```
See https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776065343

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+    const hasAttemptedLoadRef = useRef(false);
+    const currentScheduleIdRef = useRef<string | undefined>(scheduleId);
+
+    const isMobileScreen = useMediaQuery(theme.breakpoints.down('sm'));
```
```suggestion
    const isMobileScreen = useIsMobile();
```
and `const theme = useTheme();` (above) can be deleted

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+    const [error, setError] = useState<string | null>(null);
+    const [scheduleName, setScheduleName] = useState<string | null>(null);
+    const hasAttemptedLoadRef = useRef(false);
+    const currentScheduleIdRef = useRef<string | undefined>(scheduleId);
```
Small nitpick: Might be just personal preference, but I think adding some whitespace between "categories" of hooks is more readable, e.g. adding empty lines between misc hooks, store hooks, `useState`, and `useRef`.

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+    return (
+        <LocalizationProvider dateAdapter={AdapterDateFns}>
+            <PatchNotes />
+            <InstallPWABanner />
```
We should find a way to de-duplicate these components (and `NotificationSnackbar` and `HelpMenu` below) since they're copy-pasted from `Home`. Maybe a wrapper component or something?

Choollol commented on apps/antalmanac/src/routes/SharedSchedulePage.tsx:
```diff
@@ -0,0 +1,414 @@
+            <HelpMenu />
+        </LocalizationProvider>
+    );
+}
```
I think a lot of this component could be separated out into another component, maybe `SharedScheduleBanner.tsx`. It might also help with performance since it wouldn't be rerendering the entire page when state changes.
Here's what a mockup of the new `SharedSchedulePage` could look like:
```ts
export function SharedSchedulePage() {
    const isMobileScreen = useIsMobile();

    const [error, setError] = useState<string | null>(null);

    if (error) {
        return (
            <LocalizationProvider dateAdapter={AdapterDateFns}>
                <Stack component="main" height="100dvh" justifyContent="center" alignItems="center" spacing={2}>
                    <SharedScheduleBanner error={error} setError={setError} />
                </Stack>
            </LocalizationProvider>
        );
    }

    return (
        <LocalizationProvider dateAdapter={AdapterDateFns}>
            <PatchNotes />
            <InstallPWABanner />

            <Stack component="main" height="100dvh">
                <Header />
                <SharedScheduleBanner error={error} setError={setError} />
                {isMobileScreen ? <MobileSharedSchedule /> : <DesktopSharedSchedule />}
            </Stack>

            <NotificationSnackbar />
            <HelpMenu />
        </LocalizationProvider>
    );
}
```

And the new component could look something like this:
```ts
const SharedScheduleBanner = ({error, setError}: Props) => {
    const theme = useTheme();
    const navigate = useNavigate();
    const { scheduleId } = useParams<{ scheduleId: string }>();
    const { sessionIsValid } = useSessionStore();
    const { setOpenLoadingSchedule } = scheduleComponentsToggleStore();
    const [scheduleName, setScheduleName] = useState<string | null>(null);
    const hasAttemptedLoadRef = useRef(false);
    const currentScheduleIdRef = useRef<string | undefined>(scheduleId);

    const isMobileScreen = useMediaQuery(theme.breakpoints.down('sm'));

    useEffect(() => {
        if (currentScheduleIdRef.current !== scheduleId) {
            hasAttemptedLoadRef.current = false;
            currentScheduleIdRef.current = scheduleId;
        }

        if (hasAttemptedLoadRef.current) {
            return;
        }

        const loadSharedSchedule = async () => {
            if (!scheduleId) {
                setError('Invalid schedule ID');
                setOpenLoadingSchedule(false);
                hasAttemptedLoadRef.current = true;
                return;
            }

            hasAttemptedLoadRef.current = true;

            try {
                setOpenLoadingSchedule(true);

                removeLocalStorageUnsavedActions();

                if (AppStore.getSkeletonMode()) {
                    AppStore.exitSkeletonMode();
                }

                const sharedSchedule = await trpc.userData.getSharedSchedule.query({
                    scheduleId,
                });

                const scheduleSaveState = {
                    schedules: [
                        {
                            id: sharedSchedule.id,
                            scheduleName: sharedSchedule.scheduleName,
                            scheduleNote: sharedSchedule.scheduleNote || '',
                            courses: sharedSchedule.courses,
                            customEvents: sharedSchedule.customEvents,
                        },
                    ],
                    scheduleIndex: 0,
                };

                const loadSuccess = await AppStore.loadSchedule(scheduleSaveState);

                if (!loadSuccess) {
                    throw new Error('Failed to load shared schedule');
                }
                if (AppStore.getCurrentScheduleIndex() !== 0) {
                    AppStore.changeCurrentSchedule(0);
                }
                AppStore.skeletonMode = true;
                AppStore.emit('skeletonModeChange');
                setScheduleName(sharedSchedule.scheduleName);
                setOpenLoadingSchedule(false);
                setError(null);
            } catch (err) {
                console.error('Error loading shared schedule:', err);
                setError('Failed to load shared schedule. It may not exist or may have been deleted.');
                setOpenLoadingSchedule(false);
            }
        };

        loadSharedSchedule();

        return () => {
            setOpenLoadingSchedule(false);
        };
    }, [scheduleId, setOpenLoadingSchedule]);

    const handleExitSharedSchedule = useCallback(async () => {
        try {
            setOpenLoadingSchedule(true);

            if (AppStore.getSkeletonMode()) {
                AppStore.exitSkeletonMode();
            }

            if (sessionIsValid) {
                const sessionToken = useSessionStore.getState().session;

                if (sessionToken) {
                    const userDataResponse = await trpc.userData.getUserDataWithSession.query({
                        refreshToken: sessionToken,
                    });
                    const scheduleSaveState =
                        (userDataResponse as { userData?: ScheduleSaveState }).userData ?? userDataResponse;

                    if (scheduleSaveState) {
                        await AppStore.loadSchedule(scheduleSaveState as ScheduleSaveState);
                    }
                }
            } else {
                const defaultTerm = getDefaultTerm();
                const emptyScheduleData = {
                    scheduleName: defaultTerm.shortName.replaceAll(' ', '-'),
                    courses: [] as ShortCourse[],
                    customEvents: [] as RepeatingCustomEvent[],
                    scheduleNote: '',
                } as ShortCourseSchedule;
                const emptySchedule: ScheduleSaveState = {
                    schedules: [emptyScheduleData],
                    scheduleIndex: 0,
                };
                await AppStore.loadSchedule(emptySchedule);
            }

            setOpenLoadingSchedule(false);
            navigate('/');
        } catch (err) {
            console.error('Error exiting shared schedule:', err);
            setOpenLoadingSchedule(false);
            navigate('/');
        }
    }, [navigate, sessionIsValid, setOpenLoadingSchedule]);

    const handleLoadSchedule = useCallback(async (sessionToken?: string) => {
        if (sessionToken) {
            const userDataResponse = await trpc.userData.getUserDataWithSession.query({
                refreshToken: sessionToken,
            });
            const scheduleSaveState =
                (userDataResponse as { userData?: ScheduleSaveState }).userData ?? userDataResponse;

            if (scheduleSaveState) {
                await AppStore.loadSchedule(scheduleSaveState as ScheduleSaveState);
            }
        } else {
            const defaultTerm = getDefaultTerm();
            const emptyScheduleData = {
                scheduleName: defaultTerm.shortName.replaceAll(' ', '-'),
                courses: [] as ShortCourse[],
                customEvents: [] as RepeatingCustomEvent[],
                scheduleNote: '',
            } as ShortCourseSchedule;
            const emptySchedule: ScheduleSaveState = {
                schedules: [emptyScheduleData],
                scheduleIndex: 0,
            };
            await AppStore.loadSchedule(emptySchedule);
        }
    }, []);

    const handleAddToMySchedules = useCallback(async () => {
        if (!scheduleId) {
            return;
        }

        try {
            setOpenLoadingSchedule(true);

            if (AppStore.getSkeletonMode()) {
                AppStore.exitSkeletonMode();
            }

            if (sessionIsValid) {
                const sessionToken = useSessionStore.getState().session;
                if (!sessionToken) {
                    throw new Error('No session token available');
                }

                await handleLoadSchedule(sessionToken);
                await importSharedScheduleById(scheduleId);
            } else {
                const currentSchedules = AppStore.schedule.getScheduleAsSaveState();
                const sharedSchedule = await trpc.userData.getSharedSchedule.query({ scheduleId });
                const currentSchedule = currentSchedules.schedules[currentSchedules.scheduleIndex];

                if (currentSchedule && currentSchedule.scheduleName === sharedSchedule.scheduleName) {
                    const prefixedName = `(shared)-${sharedSchedule.scheduleName}`;
                    AppStore.renameSchedule(currentSchedules.scheduleIndex, prefixedName);
                    openSnackbar(
                        'success',
                        `Shared schedule "${sharedSchedule.scheduleName}" added to your schedules!`
                    );
                } else {
                    await importSharedScheduleById(scheduleId);
                }
            }

            setOpenLoadingSchedule(false);
            navigate('/');
        } catch (err) {
            console.error('Error adding schedule to account:', err);
            setOpenLoadingSchedule(false);
            if (AppStore.getSkeletonMode()) {
                AppStore.exitSkeletonMode();
            }
            navigate('/');
        }
    }, [scheduleId, sessionIsValid, navigate, setOpenLoadingSchedule, handleLoadSchedule]);

    const handleGoHome = useCallback(async () => {
        try {
            setOpenLoadingSchedule(true);

            if (AppStore.getSkeletonMode()) {
                AppStore.exitSkeletonMode();
            }

            const sessionToken = useSessionStore.getState().session;

            if (!sessionToken) {
                throw new Error('No session token available');
            }

            await handleLoadSchedule(sessionToken);

            setOpenLoadingSchedule(false);
            navigate('/');
        } catch (err) {
            console.error('Error loading user data:', err);
            setOpenLoadingSchedule(false);
            navigate('/');
        }
    }, [navigate, setOpenLoadingSchedule, handleLoadSchedule]);

    if (error) {
        return (
            <>
                <Alert severity="error">{error}</Alert>
                <Button variant="contained" onClick={handleGoHome}>
                    Go Home
                </Button>
            </>
        );
    }

    return (
        <Box
            sx={{
                px: 2,
                py: isMobileScreen ? 0.5 : 1,
                backgroundColor: 'background.paper',
                borderBottom: 1,
                borderColor: 'divider',
            }}
        >
            <Stack
                direction={'row'}
                justifyContent="space-between"
                alignItems={'center'}
                spacing={isMobileScreen ? 1 : 0}
            >
                <Typography
                    variant="h6"
                    component="h1"
                    sx={{
                        fontSize: isMobileScreen ? '1rem' : undefined,
                        overflow: 'hidden',
                        textOverflow: 'ellipsis',
                        display: 'flex',
                        alignItems: 'center',
                    }}
                >
                    Viewing Shared Schedule:{' '}
                    <strong style={{ whiteSpace: 'nowrap', marginLeft: '0.25rem' }}>{scheduleName}</strong>
                </Typography>
                <Stack direction="row" spacing={1} alignItems="center" sx={{ flexShrink: 0 }}>
                    {isMobileScreen ? (
                        <IconButton
                            aria-label="Add to My Schedules"
                            onClick={handleAddToMySchedules}
                            size="small"
                            sx={{
                                backgroundColor: 'primary.dark',
                                color: 'white',
                                borderRadius: 2,
                                '&:hover': {
                                    backgroundColor: 'primary.main',
                                },
                            }}
                        >
                            <Add />
                        </IconButton>
                    ) : (
                        <Button variant="contained" onClick={handleAddToMySchedules} size="large">
                            Add to My Schedules
                        </Button>
                    )}
                    <IconButton
                        aria-label="Exit shared schedule"
                        onClick={handleExitSharedSchedule}
                        color="inherit"
                        size={isMobileScreen ? 'medium' : 'large'}
                    >
                        <Close />
                    </IconButton>
                </Stack>
            </Stack>
        </Box>
    );
};
```
Note that `setError(null);` is moved down in `loadSharedSchedule` to prevent infinite rerenders.

You don't have to make these exact changes of course, just giving you a potential starting spot.

Choollol reviewed (changes requested):
Sorry for the comment barrage haha they're mostly small suggestions. There are a few bigger/more general issues though. I'll try to list them here:
1. https://github.com/icssc/AntAlmanac/pull/1357/changes#r2775883412
2. https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776030275
3. https://github.com/icssc/AntAlmanac/pull/1357/changes#r2776665863
4. Do we want to log analytics?
5. Something I noticed is that saving a schedule changes its ID, which means the share schedule link becomes invalid. I have no idea how we would get around this, maybe we don't have to, but I wanted to bring it up.

brelieu05 committed:
refactor: Extract schedule ID update logic into a new `setSchedules` method in `AppStore`.

brelieu05 committed:
refactor: extract schedule name into a shared utility

brelieu05 committed:
refactor: shared and imported const

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/ScheduleSelect/ScheduleSelect.tsx:
```diff
@@ -116,11 +119,14 @@ export function SelectSchedulePopover() {
     const scheduleMappingToUse = skeletonMode ? skeletonScheduleMapping : scheduleMapping;
+    const displayName = isSharedSchedulePage
+        ? AppStore.getScheduleNames()[currentScheduleIndex] || scheduleMappingToUse[currentScheduleIndex]?.name
+        : scheduleMappingToUse[currentScheduleIndex]?.name;
```
When logged in, the schedule name on the schedule select button shows the wrong schedule name:

<img width="424" height="257" alt="Image" src="https://github.com/user-attachments/assets/5e33ee39-d516-473d-a006-13741d4d3850" />

brelieu05 committed:
fix: replace console error with snackbar for schedule loading failures

Choollol committed:
Sharable course schedule links sub-branch changes (#1492)

brelieu05 committed:
fix: enable user to navigate to map and not leave sharedschedulepage

brelieu05 committed:
fix: changed clearScheduleButton from no-render to disabled for consistency

brelieu05 committed:
refactor: removed unnecessary skeletonmode on SharedScheduleBanner and refactored calendartoolbar accordingly

brelieu05 committed:
refactor: remove nullish type wfrom skeletonSchedule

brelieu05 committed:
refactor: changed deleteCourse button from no-render to disabled

brelieu05 committed:
feat: added notification to user if session isn't valid

Choollol commented on apps/antalmanac/src/components/buttons/MapLink.tsx:
```diff
@@ -10,14 +10,18 @@ interface MapLinkProps {
 export const MapLink = ({ buildingId, room }: MapLinkProps) => {
     const { setActiveTab } = useTabStore();
+    const { pathname } = useLocation();
+    const isSharedPath = pathname.startsWith('/share/');
```
This looks like duplicated logic from `useIsSharedSchedulePage`. If it's possible that this code and `useIsSharedSchedulePage` would need to change independently of each other, then it's fine to leave as is. If it's for the exact same purpose, then replacing this with a call to `useIsSharedSchedulePage` would be better.

Choollol commented on apps/antalmanac/src/components/buttons/MapLink.tsx:
```diff
@@ -10,14 +10,18 @@ interface MapLinkProps {
+    const { pathname } = useLocation();
+    const isSharedPath = pathname.startsWith('/share/');
+
+    const toLocation = isSharedPath ? `${pathname}?location=${buildingId}` : `/map?location=${buildingId}`;
```
Should we instead check if we're on any page that isn't the home page instead? It would let us not have to change anything if we add a new page later. I'm not too familiar with this area though, so final call is up to you.

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/CalendarToolbar.tsx:
```diff
@@ -198,16 +198,16 @@ export const CalendarToolbar = memo((props: CalendarPaneToolbarProps) => {
                 <Box display="flex" flexDirection="row" gap={0.5}>
                     <Box display="flex" flexWrap="nowrap" alignItems="center" gap={0.5}>
-                        <IconButton onClick={handleUndo(postHog)} disabled={skeletonMode}>
+                        <IconButton onClick={handleUndo(postHog)} disabled={isReadonlyView}>
```
Could you explain why this button (and the others) aren't disabled during skeleton mode anymore? I'm not too familiar with skeleton mode and when/how it works, but to me it seems like they should still be disabled.

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/CalendarToolbar.tsx:
```diff
@@ -284,24 +284,22 @@ export const CalendarToolbar = memo((props: CalendarPaneToolbarProps) => {
+                    <ClearScheduleButton
+                        size="medium"
+                        fontSize="small"
+                        skeletonMode={isReadonlyView}
```
https://github.com/icssc/AntAlmanac/pull/1357#discussion_r2779628556
> Could you explain why this button (and the others) aren't disabled during skeleton mode anymore? I'm not too familiar with skeleton mode and when/how it works, but to me it seems like they should still be disabled.

Choollol commented on apps/antalmanac/src/components/Calendar/Toolbar/CalendarToolbar.tsx:
```diff
@@ -198,16 +198,16 @@ export const CalendarToolbar = memo((props: CalendarPaneToolbarProps) => {
 
                         <Tooltip title="More options">
-                            <IconButton onClick={handleMenuOpen} disabled={skeletonMode}>
+                            <IconButton onClick={handleMenuOpen} disabled={isReadonlyView}>
```
There are some actions in this menu that can still be done during readonly mode, like download calendar and taking a screenshot. Maybe we could disable the edit actions in the menu rather than disable the entire menu?

btw don't worry about this one, I'll make this change.

Choollol reviewed (changes requested):
Left some questions about the new changes! I'll also be opening another PR with some more fixes/readonly-disabling.

Some important stuff:
1. The wrong schedule is showed when logged in. The schedule select popover also shows the user's personal schedules. I can try to fix this, but you might know more about it than I do.
2. How should the course popover quick search be handled? Right now, it takes the user to their home page which is fine, but the popover's position gets messed up. If this should be allowed, I can fix this in my PR.
<img width="189" height="102" alt="Image" src="https://github.com/user-attachments/assets/19d7df6c-8bc8-4dff-95ac-4aec3460b2c2" /> 

3.
>It's an intended feature as we want it as a one-time copy only rather than a true share.

👍 We should make sure users know this, maybe through a note in the shared schedule error display?

Choollol committed:
More schedule link sharing sub-branch changes (#1495)

Co-authored-by: Brendan Lieu <brendanlieu05@gmail.com>

brelieu05 committed:
refactor: replace share location heck with `useIsSharedSchedulePage` hook

brelieu05 committed:
refactor: Disable calendar toolbar buttons, including undo, redo, clear schedule, and add course, when in skeleton mode.

brelieu05 committed:
refactor: update map link to use home page path check instead of shared schedule page

brelieu05 committed:
refactor: abstract disable variable for calendar action buttons

brelieu05 committed:
fix: skip schedule load on shared pages and explicitly load user data before importing shared schedules

brelieu05 committed:
feat: disable quick search button in read-only view

brelieu05 committed:
feat: add shared schedule analytics category

brelieu05 committed:
feat: posthog log copy schedule events

brelieu05 committed:
feat: posthog log analytics event when a shared schedule is opened

brelieu05 committed:
feat: posthog logging for add shared schedule

brelieu05 committed:
fix: eslint `useEffect` and `useCallback` dependency arrays in Signin and SharedScheduleBanner components

Choollol committed:
Even more sharable course schedule sub-branch changes (#1499)

Choollol commented on apps/antalmanac/src/components/buttons/MapLink.tsx:
```diff
@@ -12,9 +11,9 @@ interface MapLinkProps {
-    const isSharedPath = useIsSharedSchedulePage();
 
-    const toLocation = isSharedPath ? `${pathname}?location=${buildingId}` : `/map?location=${buildingId}`;
+    const isHomePage = pathname === '/' || pathname == '/added' || pathname == '/map';
```
Optional: Could put this into a new custom hook, but since this logic is used only here, I think it's fine for now.

Choollol commented on apps/antalmanac/src/components/SharedScheduleBanner.tsx:
```diff
@@ -0,0 +1,316 @@
+            const sharedSchedule = await trpc.userData.getSharedSchedule.query({ scheduleId });
+
+            if (sessionIsValid) {
+                const sessionToken = useSessionStore.getState().session;
+                if (sessionToken) {
+                    const userDataResponse = await trpc.userData.getUserDataWithSession.query({
+                        refreshToken: sessionToken,
+                    });
+
+                    if (userDataResponse?.userData) {
+                        await AppStore.loadSchedule(userDataResponse.userData);
+                    }
+                }
```
Optional: Tiny bit of duplication with `handleLoadSchedule`, but if it's stable enough to be fine then I'm okay with it.

brelieu05 replied:
Yeah this was to fix the bug regarding the schedule name mismatch

Choollol reviewed (dismissed):
LGTM!

Left a couple of suggestions, but they're super tiny and optional.

alexespejo reviewed (changes requested):
Couple of things 

issue: We should have AANTS disabled
<img width="225" height="641" alt="Image" src="https://github.com/user-attachments/assets/d8834567-d3a8-49d2-9445-077fba298dfe" />

issue: Adding schedules doesn't seem to work..? When I open a shared schedule link I'm signed out. And when I hit "add to schedule" it signs me back in but it doesn't import it 

<img width="292" height="60" alt="Image" src="https://github.com/user-attachments/assets/75d45a30-6c95-4b8b-bf95-ccce7c9a4209" />

Choollol committed:
Sharable schedule link fixes (#1501)

Choollol committed:
Fix typo in HomePageWrapper docs

alexespejo reviewed (dismissed):
Send it

Choollol reviewed (dismissed):
does this work

Choollol committed:
Remove literal backticks in error snackbar

Choollol committed:
Add type safety to schedule select's anchor element

alexespejo commented on apps/antalmanac/src/backend/routers/userData.ts:
```diff
@@ -306,6 +306,41 @@ const userDataRouter = router({
+     * @param input - An object containing the userId and scheduleName.
+     * @returns The schedule ID, or throws NOT_FOUND if not found.
+     */
+    getScheduleIdByName: procedure
```
> Anyone who knows (or guesses) a userId and scheduleName

Will likely never happen

Choollol committed:
Merge branch 'main' of https://github.com/icssc/AntAlmanac into 1346-sharable-course-schedule-links
